PF/PF-Assignment-1/ShoppingCart/shoppingCart.java

```diff
@@ -0,0 +1,106 @@
+//Class to add Product along with details in the cart
+
+package ShoppingCart;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.Scanner;
+
+
+public class shoppingCart {
+	 static Scanner sc = new Scanner(System.in);
+	 ArrayList<Object> cart = new ArrayList<Object>();
+	
+/**
+ * Function to set Item Details and Add to Cart	
+ * @param index
+ */
+        void setItemDetailsAndAdd(int index){
+    	   
+    	   Product product = new Product();
+    	  
+    	   System.out.println("Enter Item : ");
+    	   product.setItem(sc.next());
+    	   
+    	   System.out.println("Enter Price : ");
+    	   product.setPrice(sc.nextInt());
+    	   
+    	   System.out.println("Enter Quantity : ");
+    	   product.setQuantity(sc.nextInt());
+    	   
+    	   cart.add(index,product);
+       }
+ 
+/**
+ * Function to remove product with details
+ * @param index
+ */
+       void removeItem(int index){
+    	   cart.remove(index);
+       }
+ 
+/**
+ * Function to show Product along with details
+ */
+       void show(){
+    	   Iterator<Object> iterator= cart.iterator(); 
+    	   int i=0;
+    	   while(iterator.hasNext()){
+    		
+    		  Product product = (Product)cart.get(i);
+    		  
+    		  System.out.println("Product Index : "+i);
+    		  System.out.println("Item : "+product.getItem());
+    		  System.out.println("Price : "+product.getPrice());
+    		  System.out.println("Quantity : "+product.getQuantity());
+    		  
+    		  System.out.println("------------------------------------");
+    		  
+    		  i++;
+    		  iterator.next();
+    		  
+    	   }
+       }
+       
+   public static void main(String args[]){
+	  
+	
+	  
+	   int choice,i=0;
+	   shoppingCart obj = new shoppingCart();
+	   
+	   do{
+			System.out.println("1. Add Product and Details");
+			System.out.println("2. Remove Product and Details");
+			System.out.println("3. Show items in the cart");
+			System.out.println("4. EXIT");
+			

```

Comment:
As per assignment -- "The items should be updatable (e.g., to change the quantity of an item in the cart,"
Current code doesn't have that. You should be able to update a item's quantity. Add an option for that.